src/shared/utils/diagnose.ts

```diff
@@ -90,25 +85,24 @@ export const diagnose = ({
     return handleError(debug, "The document must be an object");
   }
 
-  const versionToSchemaId: Record<Version, SchemaId> = {
+  const versionToSchemaId: Record<Version, SchemaId | undefined> = {
     "2.0": SchemaId.v2,
     "3.0": SchemaId.v3,
-    "4.0": SchemaId.v4,
+    "4.0": undefined,
```

Comment:
Should we add a comment to say why this is undefined?

```text
// OA v4.0 no longer uses JSON Schema for schema validation as it has all been collasped into v4Diagnose()
```

Or should the `validateSchema()` function be updated to be "smarter"?